src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```
@@ -677,43 +681,49 @@ public function getEditShipmentForm(int $orderId, Request $request): Response
+            $targetShipmentId = $targetShipment->getId();
 
-        $submittedData = $request->request->all('merge_shipment');
-        $selectedProducts = [];
+            $submittedData = $request->request->all('merge_shipment');
+            $selectedProducts = [];
 
-        foreach ($submittedData as $key => $value) {
-            if (str_starts_with($key, 'product_') && $value) {
-                $orderDetailId = (int) str_replace('product_', '', $key);
-                $quantityKey = 'quantity_' . $orderDetailId;
+            foreach ($submittedData as $key => $value) {
+                if (str_starts_with($key, 'product_') && $value) {
+                    $orderDetailId = (int) str_replace('product_', '', $key);
+                    $quantityKey = 'quantity_' . $orderDetailId;
 
-                $selectedProducts[] = [
-                    'id_order_detail' => $orderDetailId,
-                    'quantity' => (int) $submittedData[$quantityKey],
-                ];
+                    $selectedProducts[] = [
+                        'id_order_detail' => $orderDetailId,
+                        'quantity' => (int) $submittedData[$quantityKey],
+                    ];
+                }
             }
-        }
 
-        $command = new MergeProductsToShipment(
-            $shipmentId,
-            $targetShipmentId,
-            $selectedProducts
-        );
-        $this->dispatchCommand($command);
+            $command = new MergeProductsToShipment(
+                $shipmentId,
+                $targetShipmentId,
+                $selectedProducts
+            );
+            $this->dispatchCommand($command);
+        } catch (Exception $e) {
+            if ((new ReflectionClass($e))->getShortName() === 'CannotEditShipmentShippedException') {
+                $this->addFlash('error', $this->trans('You cannot split shipment when is already shipped.', [], 'Admin.Orderscustomers.Notification'));
+            }
```

You should rethrow the exceptions you don't handle

```suggestion
            } else {
                throw $e;
            }
```